dpareto/data/adult/importer.py

```diff
@@ -7,45 +7,8 @@
 
 def import_adult_dataset(data_ctx):
     try:
-        #print("Unpickling...")
         Xtrain, Ytrain = pickle.load( open( "dpareto/data/adult/a1a.train.p", "rb" ) )
         Xtest, Ytest = pickle.load( open( "dpareto/data/adult/a1a.test.p", "rb" ) )
-        #print("unpickled.")
         return Xtrain, Ytrain, Xtest, Ytest
     except FileNotFoundError as _:
-        print("unpickling failed.")
-        print("Opening...")
-        with open("dpareto/data/adult/a1a.train") as f:
-            train_raw = f.read()
-
-        with open("dpareto/data/adult/a1a.test") as f:
-            test_raw = f.read()
-        print("done opening.")
-
-        def process_data(raw_data):
-            train_lines = raw_data.splitlines()
-            num_examples = len(train_lines)
-            num_features = 123
-            X = nd.zeros((num_examples, num_features), ctx=data_ctx)
-            Y = nd.zeros((num_examples, 1), ctx=data_ctx)
-            for i, line in enumerate(train_lines):
-                tokens = line.split()
-                label = (int(tokens[0]) + 1) / 2  # Change label from {-1,1} to {0,1}
-                Y[i] = label
-                for token in tokens[1:]:
-                    index = int(token[:-2]) - 1
-                    X[i, index] = 1
-            return X, Y
-
-        print("Processing...")
-        Xtrain, Ytrain = process_data(train_raw)
-        Xtest, Ytest = process_data(test_raw)
-        print("done processing.")
-
-        make_pickle = False
-        if make_pickle:
-            print("Pickling...")
-            pickle.dump( (Xtrain, Ytrain), open( "dpareto/data/adult/a1a.train.p", "wb" ) )
-            pickle.dump( (Xtest, Ytest), open( "dpareto/data/adult/a1a.test.p", "wb" ) )
-            print("pickled.")
-        return Xtrain, Xtest, Ytrain, Ytest
+        print('Adult dataset files not found. Try running the downloader from the project root.')
```

Comment:
Any reason not to directly attempt the download? That's what e.g. mxnet does right?

Reply:
MXNet provides some pre-defined methods for fetching certain datasets (such as MNIST) here: https://beta.mxnet.io/_modules/mxnet/gluon/data/vision/datasets.html
However, it doesn't have pre-defined support for fetching the Adult dataset. We could modify their approach to download the data, but this approach is a quicker and more-explicit way (although arguably dirtier). This approach can also act as an example for users of the project who want to use their own datasets.

Reply:
Oh I didn't mean to use their downloader, but rather provide the same experience (i.e. check if it's there, otherwise attempt to download), so that the user doesn't have to go and run an extra script.

Reply:
Ah, I see. This approach simplified implementation decisions that would otherwise need to be made for when/where the checking and downloading would need to be done. Specifically, it would be easy to implement this if the user just tried to train 1 model on the Adult dataset. However, if the user decided to run the full DPareto pipeline with multiprocessing, then each individual process would naively try to download and process it's own Adult dataset. To defend against that sort of thing, we'd need to do more engineering work than it's worth.